### PR TITLE
map analysis

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR is a mean average precision analysis of the cell death classes

### CODE REVIEW

**Review by axiomcura — APPROVED** [review]
LGTM, left some comments ! 

Nice to see mAP begging applied

File: 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb [hunk 1]
```diff
@@ -0,0 +1,954 @@
+{
```
Comment (axiomcura) [inline 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    df_feats = np.array(df_feats)
</pre><p>Confirming: Roshan's code returns both a pd.DataFrame for metadata and np.array for feature values. Is there a specific reason for using 'df_feats'  in <code>np.array(df_feats)</code>? I want to ensure I'm interpreting this correctly.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (axiomcura) [inline 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb:1]: <p>Something looks off about the p-values. It's alternating between <code>0.5</code> and <code>1,0</code> Considering revising the generated AP scores and see if that pattern is reoccurring. If you have an explanation of this, please feel free to disregard this comment.  </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (axiomcura) [inline 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #18.    df_feats = np.array(df_feats)
</pre><p>similar comment above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (axiomcura) [inline 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb:1]: <p>Same comment about the P-values</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (axiomcura) [inline 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.    df_feats = np.array(df_feats)
</pre><p>similar comment</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

File: 9.mAP/notebooks/1.aggregate_map_scores_class_level.ipynb [hunk 2]
```diff
@@ -0,0 +1,701 @@
+{
```
Comment (axiomcura) [inline 9.mAP/notebooks/1.aggregate_map_scores_class_level.ipynb:1]: <p>The comment above does not reflect what is occurring in this block. Consider removing comments <code>lines: 1-5</code> in order to improve readability </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (axiomcura) [inline 9.mAP/notebooks/1.aggregate_map_scores_class_level.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # aggregate single cells scores with cell UUID
</pre><p>Update this <code>aggregating single cells using Metadata Well</code></p><p>In order to improve clarity </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (axiomcura) [inline 9.mAP/notebooks/1.aggregate_map_scores_class_level.ipynb:1]: <p>It seems that the p-value pattern of <code>0.5 and 1.0</code> is still here. If this  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/WayScience/pyroptosis_signature_data_analysis/pull/112/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

File: 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb (hunk 1 above)
Comment (MikeLippincott) [reply 9.mAP/notebooks/0.generate_map_scores_class_level.ipynb:1]: <p>I was getting an error so I converted to numpy array...interesting that I got the error</p>

**Comment by MikeLippincott** [comment]
All comments addressed and I fixed the p-value "wonkiness" by changing the null_distribution size :)

**Pull request merged**